### DISCUSSION TITLE
Dealing with spctl for plugins

### DISCUSSION THREAD

**Discussion by rbdr** [quicksilver/Quicksilver#2947] [Q&A] [answered]
Hi Everyone!

I had a question that perhaps y'all might be familiar with: What's the recommended way to avoid spctl blocking plugins? Do you actually submit the plugins to the notarization process?

I just built a UI plugin (https://git.sr.ht/~rbdr/Flat-Bezel.qsplugin / https://unlimited.pizza/flat_bezel.html) and sign it with `codesign --force -vvv --deep --sign` using a Developer ID certificate.

This works well locally, but as soon as I package it and upload / dowlnoad again, gatekeeper blocks it until I give permissions in "Security & Privacy". Since this is a bit annoying for end-users, I was wondering what the recommended approach is.

My guess is to run through the notarization process, but wanted to check in case there's other recommendations from the community.

**Comment by n8henrie** [comment, marked as answer]
> Do you actually submit the plugins to the notarization process?

No, signed but not notarized. The whole process is done in CI, so you can compare / contrast the process for [Calculator-qsplugin](https://github.com/quicksilver/Calculator-qsplugin/blob/fff763e310155e4d9fdf32436f399bcaa6584c92/.github/workflows/ci.yml) vs [Quicksilver.app](https://github.com/quicksilver/Quicksilver/blob/ef9e74c3c1174391dfb331ba188878fe9a2a4470/.github/workflows/ci.yml) with the latter including calls to [qssign](https://github.com/quicksilver/Quicksilver/blob/ef9e74c3c1174391dfb331ba188878fe9a2a4470/Quicksilver/Tools/qssign) with the `xcrun notarytool` parts.

> This works well locally, but as soon as I package it and upload / dowlnoad again, gatekeeper blocks it until I give permissions in "Security & Privacy".

Some relevant conversations:

- https://github.com/quicksilver/Quicksilver/issues/2868
- https://github.com/quicksilver/Quicksilver/issues/2576
- https://github.com/quicksilver/Quicksilver/issues/2654
- https://github.com/quicksilver/Quicksilver/pull/2655

Sorry I haven't really curated those for you.

Don't quote me on this, but IIRC we have the same issue with manually downloaded plugins, see [here](https://github.com/quicksilver/Quicksilver/issues/2634#issuecomment-1073045396) and [here](https://github.com/quicksilver/Quicksilver/issues/2634#issuecomment-1073292434).

**Reply by rbdr** [reply]
This is super valuable, I went over part of the conversations and I'll take a deeper look soon. but I think the last two especially answer my question.

Thank you for the help! (And for the really quick response :D)
